### PR TITLE
LogoVote : add registLogo testcase

### PR DESCRIPTION
**Pull request by y12studio**

### CODE REVIEW

File: LogoVote/truffle/contracts/LogoVote.sol [hunk 1]
```diff
@@ -78,7 +77,7 @@ contract LogoVote is Pausable, SafeMath{
 		if (!msg.sender.send(amount)) throw;
 	}
 
-	function claimReward (address _receiver) stopInEmergency afterEnd {
+	function claimRewardV1 (address _receiver) stopInEmergency afterEnd {
```
Comment (y12studio) [inline LogoVote/truffle/contracts/LogoVote.sol]: @willliw  Does it really need to keep this old version ?

Comment (willliw) [reply LogoVote/truffle/contracts/LogoVote.sol]: 不用留，這些都是我自己沒merge好, sorry。

**Comment by y12studio** [comment]
@willliw @ChihChengLiang  Ready for review and merge.

File: LogoVote/truffle/test/logovote.js [hunk 2]
```diff
@@ -22,4 +22,74 @@ contract('LogoVote', function (accounts) {
...
+          logoVote.registLogo(accounts[1], accounts[2], 'http://logo1'),
+          logoVote.registLogo(accounts[2], accounts[3], 'http://logo2')
+        ])
+    }).then(function (values) {
+      assert.equal(values.length, 3)
+      return logoVote.getLogos.call()
+    }).then(function (logos) {
+      return logoVote.isLogo.call(logos[0])
+    }).then(function (_isLogoResult) {
+      assert.isOk(_isLogoResult, 'isLogo shoule be ok')
```
Comment (ChihChengLiang) [inline LogoVote/truffle/test/logovote.js]: typo, and another same typo in next assert.

File: LogoVote/truffle/test/logovote.js [hunk 3]
```diff
@@ -22,4 +22,74 @@ contract('LogoVote', function (accounts) {
       assert.equal(totalSupply.valueOf(), 10000)
     })
   })
-})
+
+  it('registLogo should be ok', function () {
```
Comment (ChihChengLiang) [inline LogoVote/truffle/test/logovote.js]: `should be ok` is too ambiguous. The sentence could tell more about what this test is doing and what the function *should behave*. So a new developer could understand the function and the test easier.
For example,
```
it('#registLogo() should increase the length of logos array from 0 to 1', ...
```

File: LogoVote/truffle/test/logovote.js [hunk 4]
```diff
@@ -22,4 +22,74 @@ contract('LogoVote', function (accounts) {
...
+    var logoVote
+    return LogoVote.deployed().then(function (instance) {
+      logoVote = instance
+      // Sending Ether / Triggering the fallback function
+      return logoVote.sendTransaction({
+        from: accounts[0],
+        value: web3.toWei(1.1, "ether")
+      })
+    }).then(function (_txr) {
+      // console.log(_txr)
```
Comment (ChihChengLiang) [inline LogoVote/truffle/test/logovote.js]: debug code should not be committed.

File: LogoVote/truffle/test/logovote.js [hunk 5]
```diff
@@ -22,4 +22,74 @@ contract('LogoVote', function (accounts) {
...
+      return logoVote.isLogo.call(logos[0])
+    }).then(function (_isLogoResult) {
+      assert.isOk(_isLogoResult, 'isLogo shoule be ok')
+      return logoVote.isLogo.call(0xdead)
+    }).then(function (_isLogoResult) {
+      assert.isNotOk(_isLogoResult, 'isLogo shoule not be ok')
+    })
+  })
+
+  it('donate should be ok', function () {
```
Comment (ChihChengLiang) [inline LogoVote/truffle/test/logovote.js]: example:
`it('should show #totalReward() 6.6 ether after #donate() 1.1, 2.2, and 3.3 ether.')`

File: LogoVote/truffle/test/logovote.js [hunk 6]
```diff
@@ -22,4 +22,74 @@ contract('LogoVote', function (accounts) {
...
+      assert.isOk(_tx0)
+      tx0 = _tx0
+      return logoVote.getLogos.call()
+    }).then(function (logos) {
+      console.log(logos)
+      assert.equal(logos.length, 1)
+    })
+  })
+
+  it('isLogo should be ok', function () {
```
Comment (ChihChengLiang) [inline LogoVote/truffle/test/logovote.js]: example:
```
it('#isLogo() should tell if an address is in logos, or not in')
```

**Review by ChihChengLiang — CHANGES_REQUESTED** [review]
Awesome works on testing major functions of the contract. Some changes can make the tests look better.

**Comment by willliw** [comment]
utACK

**Review by willliw — APPROVED** [review]

**Review by ChihChengLiang — APPROVED** [review]

**Pull request merged**